src/token/TokenReference.js

```diff
@@ -0,0 +1,57 @@
+import NftId from "./NftId.js";
+import TokenId from "./TokenId.js";
+
+/**
+ * @namespace proto
+ * @typedef {import("@hashgraph/proto").proto.TokenReference} HashgraphProto.proto.TokenReference
+ */
+
+export default class TokenReference {
+    constructor() {
+        /**
+         * @public
+         * @type {?TokenId}
+         */
+        this.fungibleToken = null;
+        /**
+         * @public
+         * @type {?NftId}
+         */
+        this.nft = null;
+        /**
+         * @public
+         * @type {?"fungibleToken"|"nft"}
+         */
+        this.tokenIdentifier = null;
+    }
+
+    /**
+     * @public
+     * @param {HashgraphProto.proto.TokenReference} reference
+     * @returns {TokenReference}
+     */
+    static _fromProtobuf(reference) {
+        /**
+         * @type {"fungibleToken"|"nft"|null}
```

Comment:
There is a `TokenType` that must be used in this case.
```suggestion
         * @type {TokenType}
```

Reply:
I removed all references that use this tokenIdentifier field as the other SDKs do.